crates/loro-internal/src/container.rs

```diff
@@ -36,18 +36,16 @@ pub mod idx {
     }
 
     impl ContainerIdx {
-        pub(crate) const TYPE_MASK: u32 = 0b1111 << 28;
+        pub(crate) const TYPE_MASK: u32 = 0b11111 << 27;
         pub(crate) const INDEX_MASK: u32 = !Self::TYPE_MASK;
 
         #[allow(unused)]
         pub(crate) fn get_type(self) -> ContainerType {
-            match (self.0 & Self::TYPE_MASK) >> 28 {
-                0 => ContainerType::Map,
-                1 => ContainerType::List,
-                2 => ContainerType::Text,
-                3 => ContainerType::Tree,
-                4 => ContainerType::MovableList,
-                _ => unreachable!(),
+            let a = (self.0 & Self::TYPE_MASK) >> 27;
+            if self.is_unknown() {
+                ContainerType::Unknown((a & 0xf) as u8)
```

Comment:
Why & 0xf? Can we extract these &0x8 &0.f to a separate function with docs?

Reply:
> Why & 0xf? 

The last four bits are the value of the type?